app/controllers/kudos_controller.rb

```diff
@@ -35,11 +40,11 @@ def update
   end
 
   def destroy
-    @kudo.destroy
-    if @kudo.destroy
+    if current_employee == @kudo.giver
+      @kudo.destroy
       flash[:notice] = 'Kudo was successfully deleted'
     else
-      flash[:alert] = 'Something went wrong delete operation, please try again'
+      flash[:alert] = 'You are not authorized to perform this operation'
     end
     redirect_to root_path
```

Comment:
what happens when `@kudo.destroy` will not work for example because of db malfunction?

Reply:
There is no consideration of failing. I will include this suggestion in the next commit.